libs/datasets/timeseries.py

```diff
@@ -41,7 +41,8 @@ def states(self) -> List:
 
     @property
     def state_data(self) -> pd.DataFrame:
-        return self.get_subset(AggregationLevel.STATE).data
+        # Return a copy instead of a slice so the caller may modify it without getting a SettingWithCopyWarning warning.
+        return self.get_subset(AggregationLevel.STATE).data.copy()
```

Comment:
I wonder if instead of copying the data, you could change all of the filters in `get_subset` to filter like so:
```
        data = self.data
        if aggregation_level:
            data = data.loc[data.aggregate_level == aggregation_level.value, :]
``

Reply:
Nice catch. I think I can do even better. Cranking on it...